### PR TITLE
FALCON-2066 : Add support for various database types for import and e…

### PR DESCRIPTION
**Pull request by vramachan**
+ Add support for various database types
+ Validate new database types work 
+ Add examples for various new database types
+ Add documentation about database types
+ Remove specifying driver option for supported database types

### CODE REVIEW

File: cli/pom.xml [hunk 1]
```diff
@@ -39,6 +39,16 @@
         <dependency>
             <groupId>org.apache.hadoop</groupId>
             <artifactId>hadoop-auth</artifactId>
+            <exclusions>
```
Comment (bvellanki) [inline cli/pom.xml]: These changes went in with Pull 218

File: client/pom.xml [hunk 2]
```diff
@@ -78,7 +78,9 @@
         <dependency>
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-api</artifactId>
+            <scope>provided</scope>
```
Comment (bvellanki) [inline client/pom.xml]: Changes went in with pull-218

File: client/pom.xml [hunk 3]
```diff
@@ -117,7 +119,6 @@
                             <goal>copy-dependencies</goal>
                         </goals>
                         <configuration>
-                            <excludeScope>provided</excludeScope>
```
Comment (bvellanki) [inline client/pom.xml]: This change went in with pull-218

**Comment by bvellanki** [comment]
These are very useful set of examples @vramachan , +1 once you remove the code that was already committed using pull-218

File: oozie/src/main/java/org/apache/falcon/oozie/DatabaseExportWorkflowBuilder.java [hunk 4]
```diff
@@ -25,7 +25,9 @@
 import org.apache.falcon.entity.FeedHelper;
 import org.apache.falcon.entity.Storage;
 import org.apache.falcon.entity.v0.cluster.Cluster;
+import org.apache.falcon.entity.v0.datasource.Credential;
```
Comment (peeyushb) [inline oozie/src/main/java/org/apache/falcon/oozie/DatabaseExportWorkflowBuilder.java]: Credential class has not been used.

File: oozie/src/main/java/org/apache/falcon/oozie/DatabaseExportWorkflowBuilder.java [hunk 5]
```diff
@@ -25,7 +25,9 @@
...
 import org.apache.falcon.entity.v0.datasource.Datasource;
+import org.apache.falcon.entity.v0.datasource.DatasourceType;
```
Comment (peeyushb) [inline oozie/src/main/java/org/apache/falcon/oozie/DatabaseExportWorkflowBuilder.java]: DatasourceType has not been used.

File: oozie/src/main/java/org/apache/falcon/oozie/DatabaseImportWorkflowBuilder.java [hunk 6]
```diff
@@ -26,6 +26,7 @@
 import org.apache.falcon.entity.Storage;
 import org.apache.falcon.entity.v0.cluster.Cluster;
 import org.apache.falcon.entity.v0.datasource.Datasource;
+import org.apache.falcon.entity.v0.datasource.DatasourceType;
```
Comment (peeyushb) [inline oozie/src/main/java/org/apache/falcon/oozie/DatabaseImportWorkflowBuilder.java]: DatasourceType class has not been used.

**Comment by yzheng-hortonworks** [comment]
+1, after fixing merge comment from @bvellanki and checkstyle comment from @peeyushb .

File: oozie/src/main/java/org/apache/falcon/oozie/DatabaseExportWorkflowBuilder.java [hunk 7]
```diff
@@ -94,12 +96,15 @@ private String buildSqoopCommand(Cluster cluster, Feed feed) throws FalconExcept
...
         StringBuilder sqoopOptions = new StringBuilder();
 
-        buildConnectArg(sqoopArgs, cluster).append(ImportExportCommon.ARG_SEPARATOR);
-        buildTableArg(sqoopArgs, cluster).append(ImportExportCommon.ARG_SEPARATOR);
-        Datasource datasource = DatasourceHelper.getDatasource(FeedHelper.getExportDatasourceName(
-                FeedHelper.getCluster(entity, cluster.getName())));
-        ImportExportCommon.buildUserPasswordArg(sqoopArgs, sqoopOptions, datasource)
-                .append(ImportExportCommon.ARG_SEPARATOR);
+        org.apache.falcon.entity.v0.feed.Cluster feedCluster = FeedHelper.getCluster(entity, cluster.getName());
+        Datasource datasource = DatasourceHelper.getDatasource(FeedHelper.getExportDatasourceName(feedCluster));
```
Comment (vrangan) [inline oozie/src/main/java/org/apache/falcon/oozie/DatabaseExportWorkflowBuilder.java]: How do we handle connector specific direct mode and verbose options from the properties (which are key value pairs but these are single arguments to Sqoop)

File: examples/entity/importexport/feed-export-hcat.xml [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+    <ACL owner="ambari-qa" group="users" permission="0755"/>
+    <schema location="/none" provider="none"/>
+
+   <lifecycle>
+        <retention-stage>
+            <frequency>days(1)</frequency>
+            <queue>default</queue>
+            <priority>LOW</priority>
+           <properties>
```
Comment (PraveenAdlakha) [inline examples/entity/importexport/feed-export-hcat.xml]: formatting for tag properties.

File: oozie/src/main/java/org/apache/falcon/oozie/DatabaseExportWorkflowBuilder.java (hunk 7 above)
Comment (vramachan) [inline oozie/src/main/java/org/apache/falcon/oozie/DatabaseExportWorkflowBuilder.java]: @vrangan Need some thinking on this and so created FALCON-2083 to track that.

File: client/pom.xml (hunk 3 above)
Comment (vramachan) [inline client/pom.xml]: @bvellanki Removed the changes from pull-218

File: oozie/src/main/java/org/apache/falcon/oozie/DatabaseImportWorkflowBuilder.java (hunk 6 above)
Comment (vramachan) [inline oozie/src/main/java/org/apache/falcon/oozie/DatabaseImportWorkflowBuilder.java]: @peeyushb fixed all checkstyles.

File: examples/entity/importexport/README.txt [hunk 9]
```diff
@@ -0,0 +1,56 @@
+####### Using MySQL as Datasource #######
+yum install mysql mysql-server mysql-libs
```
Comment (PraveenAdlakha) [inline examples/entity/importexport/README.txt]: This is OS specific and currently we support few more db instances, please mention its only specific to mysql and os details.

There are few assumptions  in the script like oozie is setup and installed... As installation directory can vary from installation to installation better get if generically  through environment variables.

Comment (PraveenAdlakha) [inline examples/entity/importexport/README.txt]: please ignore the 1st line just saw you have mentioned it in the end..just see if you can generically get the oozie installation dir, to make the script better.

File: examples/entity/importexport/feed-export-hcat.xml (hunk 8 above)
Comment (vramachan) [inline examples/entity/importexport/feed-export-hcat.xml]: formatted.

**Pull request closed without merging**